kubernetes/deployment-production.tmpl

```diff
@@ -47,6 +47,58 @@ data:
       }
 
       location / {
+        # avoid attacker setting the host value for links in rails_app
+        # https://github.com/rails/rails/issues/29893
+        proxy_set_header X-Forwarded-Host $server_name;
```

Comment:
Do we definitely want to use `$server_name` rather than `$host` here?

Reply:
Or maybe `$http_host`? (It's confusing that there are so many similar variables...)

Reply:
I wasn't 100% tbh as my reading suggests that using `$host` or `$http_host` will take it from the request line, then the http header http://nginx.org/en/docs/http/ngx_http_core_module.html#var_server_name
>$host
    in this order of precedence: host name from the request line, or host name from the “Host” request header field, or the server name matching a request 

I beleive that means it'll match the host from the request URI as per https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html

But it may fall back to http header which is the situation we're trying to avoid. I believe using `$server_name` explicitly avoids this scenario but does mean our nginx configs have to be match our DNS entries for the target service...perhaps not ideal. 

Thoughts?

Reply:
Ah, ok I get it now. I was misunderstanding what we were trying to do. If we're specifically trying not to pass the user-supplied hostname then `server_name` sounds right.